Fixing broken GLM serialization/deserialization

michaellavelle opened the pull request:
When attempting to write() and then read() a GLM instance, the following exception is thrown:

java.lang.IllegalArgumentException: Unknown link function: identityIntercept
	at mltk.predictor.LinkFunction.get(LinkFunction.java:28)
	at mltk.predictor.glm.GLM.read(GLM.java:132)

This PR corrects the format of the output of the write method and modifies the read method accordingly

yinlou commented on src/mltk/predictor/glm/GLM.java:
```diff
@@ -144,7 +145,7 @@ public void read(BufferedReader in) throws Exception {
 	public void write(PrintWriter out) throws Exception {
 		out.printf("[Predictor: %s]\n", this.getClass().getCanonicalName());
-		out.printf("Link: " + link);
+		out.println("Link: " + link);
```
Thanks for finding the bug!

yinlou commented on src/mltk/predictor/glm/GLM.java:
```diff
@@ -130,6 +130,7 @@ public double intercept(int k) {
 	public void read(BufferedReader in) throws Exception {
 		in.readLine();
```
Actually this line should be removed:)

michaellavelle commented:
Thanks for your reply @yinlou -  I've now added Javadoc to the GLM.read() method to indicate that the method is intended to be used internally and that PredictorReader.read should be used instead, and I've removed the readLine() addition from GLM.read().   In the testing I've done, it seems the fix I added to the GLM.write() method for the line break is still needed - do let me know if there are any issues with this.

Thanks,

Michael

yinlou commented:
Hi @michaellavelle, thanks for adding the comment. I think in stead of adding comments for each predictor, it's better to override the document for Predictor.read() and Predictor.write(). You can see the diff in latest version. To prevent confusion in the future, I will sketch out a wiki detailing how to work with this package. Thanks for you contribution!

michaellavelle commented:
Thanks @yinlou - I've now updated our fork with your fixes - many thanks for all your work on this project
